### PR TITLE
Add a test for MQTT will message with non-retain flag

### PR DESCRIPTION
**Pull request by orpiske**
This patch adds a new test that verifies if the broker is able to send
a will message if the retain flag is set to false.

### CODE REVIEW

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTTest.java [hunk 1]
```diff
@@ -1067,6 +1067,43 @@ public boolean isSatisfied() throws Exception {
...
+         @Override
+         public boolean isSatisfied() throws Exception {
+            return connection.isConnected();
+         }
+      });
+
+      // kill transport
+      connection.kill();
+
+      Thread.sleep(10000);
```
Comment (clebertsuconic) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTTest.java]: no no no no no no no no no!!!!!!!!!!!!!!!! :)

10 seconds on a test.. that's an eternety.. I have been trying to get rid of sleeps on the testsuite, not the opposite :)

Comment (clebertsuconic) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTTest.java]: We should rather have a Wait clause here.

Comment (mtaylor) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTTest.java]: @orpiske Any change you could change the sleep to use a latch.  There was a conditional latch added recently that will exit once a condition is met.  See the ProtonTest for an example.

Thanks

**Comment by orpiske** [comment]
@clebertsuconic / @mtaylor Sure thing. That should be fine. I basically just copied the TC above this one to reproduce the issue. I will take a look and update it accordingly.

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTTest.java (hunk 1 above)
Comment (orpiske) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTTest.java]: @clebertsuconic / @mtaylor Sure thing. That should be fine. I basically just copied the TC above this one to reproduce the issue. I will take a look and update it accordingly.

**Comment by mtaylor** [comment]
@orpiske The test is not valid.  If you are sending WILL non retained, then the consumer must be subscribed **before** the connection is killed.  I moved the kill connection statement to after the client subscription and the test passes.

**Comment by orpiske** [comment]
After some discussions w/ the developers, this code does not properly reproduce the issue raised on the C code and, among other things, is wrong. Therefore, closing the PR.

**Pull request closed without merging**